### PR TITLE
POS-130 - Fuzz tests

### PR DESCRIPTION
**Pull request by 0xSasaPrsic**
Added fuzz tests for PBFT.

### CODE REVIEW

File: e2e/fuzz_unreliable_network_test.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+					if bp == v.name { // is part of the bigPartition
+						runningMajorityNodes = append(runningMajorityNodes, v.name)
+					}
+				}
+			} else {
+				stoppedNodes = append(stoppedNodes, v.name)
+			}
+		}
+		// check all running nodes in majority partition for the block height
+		fmt.Printf("Checking for height %v, started with nodes %d\n", currentHeight, nodesCount)
```
Comment (ferranbt) [inline e2e/fuzz_unreliable_network_test.go:72]: Use `t.Log` instead of Print.

File: e2e/fuzz_unreliable_network_test.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFuzz_Unreliable_Network(t *testing.T) {
+	rand.Seed(time.Now().Unix())
```
Comment (ferranbt) [inline e2e/fuzz_unreliable_network_test.go:14]: Add a condition to only enabled this if `FUZZ` env variable is set.

**Review by ferranbt — CHANGES_REQUESTED** [review]

File: e2e/fuzz_unreliable_network_test.go (hunk 1 above)
Comment (0xSasaPrsic) [reply e2e/fuzz_unreliable_network_test.go]: Done.

File: e2e/fuzz_unreliable_network_test.go (hunk 2 above)
Comment (ferranbt) [reply e2e/fuzz_unreliable_network_test.go:18]: Add this into a function
```
func isFuzzEnabled(t *testing.T) {
        if os.Getenv("FUZZ") != "true" {
		t.Skip("Fuzz tests are disabled.")
	}
}
```

**Comment by ferranbt** [comment]
Could you enable fuzz tests in CI:
- Add a new command `make fuzz` that only runs the Fuzz tests (`--run Fuzz`)
- Add a parallel fuzz task in Github that only runs the fuzz tests with some 10m timeout.

**Review by ferranbt — APPROVED** [review]

**Pull request merged**